uvicorn/protocols/websocket.py

```diff
@@ -0,0 +1,138 @@
+import asyncio
+import websockets
+
+
+def websocket_upgrade(http):
+    request_headers = dict(http.headers)
+    response_headers = []
+
+    def get_header(key):
+        key = key.lower().encode('utf-8')
+        return request_headers.get(key, b'').decode('utf-8')
+
+    def set_header(key, val):
+        response_headers.append((key.encode('utf-8'), val.encode('utf-8')))
+
+    try:
+        key = websockets.handshake.check_request(get_header)
+        websockets.handshake.build_response(set_header, key)
+    except websockets.InvalidHandshake:
+        
+        # TODO
+
+        # http.loop.create_task(http.active_request.send({
+        #     'status': 400,
+        #     'headers': [[b'content-type', b'text/plain']],
+        #     'content': b'Invalid WebSocket handshake'
+        # }))
+        return
+
+    protocol = WebSocketProtocol(http, response_headers)
+    protocol.connection_open()
+    protocol.connection_made(http.transport, http.scope)
+    http.transport.set_protocol(protocol)
+
+
+class Request():
```

Comment:
just curious, why it is Req(): not Req: ?

Reply:
The code in the parent branch defines classes in this way, so I was following the existing convention to keep things consistent - I am not sure if there was a particular reason.